lib/internal/modules/esm/resolve.js

```diff
@@ -1154,14 +1162,14 @@ function defaultResolve(specifier, context = {}) {
  */
 function decorateErrorWithCommonJSHints(error, specifier, parentURL) {
   const found = resolveAsCommonJS(specifier, parentURL);
-  if (found) {
+  if (found && found !== specifier) {
```

Comment:
Can you add a comment to here? Why did we make this change?

Reply:
If the hint is suggesting the exact same string the user already provided, it's not a helpful hint, it's just annoying and obviously wrong. Can you suggest a comment?

Reply:
I didn't understand why we had to check `found !== specifier` but not prior to this PR. That's why I asked.

Reply:
Ah ok! The linked issue is precisely reporting that problem (sometimes the hint would suggest the exact same string the user already passed), so I thought it would be self-explanatory. Do you still think I should add a comment?

Reply:
I think a comment would be useful. It’s not obvious just from reading the code, without the context of this PR or the issue that preceded this.

```suggestion
  if (found && found !== specifier) { // Don't suggest the same input the user provided
```